tests/rados/test_osd_replicated_inconsistency_scenario.py

```diff
@@ -0,0 +1,219 @@
+            )
+            rados_obj.log_cluster_health()
+            return 1
+        result = verify_pg_state(rados_obj, pg_id)
+        if not result:
+            log.error("The  pg state output contain repair state after scrub")
+            return 1
+
+        method_should_succeed(rados_obj.delete_pool, replicated_pool_name)
+        time.sleep(5)
+        pg_id = create_pool_inconsistent_object(
+            rados_obj, no_of_objects, pool_obj, **replicated_config
+        )
+
+        obj_count = get_inconsistent_count(scrub_object, pg_id, rados_obj, "deep-scrub")
+        if obj_count == -1:
+            log.error(f"The deep-scrub not initiated on the pg-{pg_id}")
+            return 1
+
+        if obj_count != 0:
+            log.error(
+                f"deep-scrub repaired the {no_of_objects - obj_count} inconsistent objects"
+            )
+            rados_obj.log_cluster_health()
+            return 1
+        result = verify_pg_state(rados_obj, pg_id)
+        if not result:
+            log.error("The  pg state output contain repair state after deep-scrub")
+            return 1
+
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Execution of finally block")
+        if config.get("delete_pool"):
+            method_should_succeed(rados_obj.delete_pool, replicated_pool_name)
+            log.info("deleted the pool successfully")
+        set_default_param_value(mon_obj)
+
+        rados_obj.log_cluster_health()
+    return 0
+
+
+def create_pool_inconsistent_object(
+    rados_object,
+    no_of_objects,
+    pool_obj,
+    **config,
+):
+    """
+     The method create a replicated pool and generate the inconsistent objects
+    Args:
+        rados_object: Rados object
+        no_of_objects: numer of objects to convert into inconsistent objects
+        pool_obj: pool object
+        config: pool configurations
+    Returns: Returns  the pg id
+    """
+    pool_name = config["pool_name"]
+    if not rados_object.create_pool(**config):
+        log.error("Failed to create the replicated Pool")
+        return 1
+    if not pool_obj.fill_omap_entries(
+        pool_name=pool_name, obj_end=50, num_keys_obj=100
+    ):
+        log.error(f"Omap entries not generated on pool {pool_name}")
+        raise Exception(f"Omap entries not generated on pool {pool_name}")
+    rep_obj_list = rados_object.get_object_list(pool_name)
+    rep_count = 0
+    for obj in rep_obj_list:
+        try:
+            # Create inconsistency objects
+            try:
+                pg_id = rados_object.create_inconsistent_object(pool_name, obj)
+            except Exception as err:
+                log.info(
+                    f"Cannot able to convert the object-{obj} into inconsistent object."
+                    f"Picking another object to convert-{err}"
+                )
+                continue
```

Comment:
We are not checking how many failures have occurred, and if no object could be corrupted to generate inconsistency.

In worst cases, we could end up with 0 inconsistent objects. We need to add some check/count to see how many are generated.

Reply:
I added the statement about how many objects are converted into inconsistent objects.